ArduCopter/mode_guided.cpp

```diff
@@ -313,14 +313,11 @@ void ModeGuided::angle_control_start()
 
     // initialise targets
     guided_angle_state.update_time_ms = millis();
-    guided_angle_state.attitude_quat.initialise();
+    guided_angle_state.attitude_quat.from_euler(Vector3f(0.0, 0.0, attitude_control->get_att_target_euler_rad().z));
```

Comment:
I might be wrong about it's impact, but attitude quaternion is also initialized in the `angle_control_run` method where timeout check happens. Maybe it should be adjusted there as well

Reply:
Ok, I did the test I my assumption turned out to be valid. After initial switch to GUIDED_NOGPS the vehicle keeps it's current orientation, but after 3 seconds of inactivity automatically rotates north. 
Applying the change in both places does resolve the problem

Reply:
Nice catch!. I will add that to the pr. Thank you very much!!

And thanks for testing this.